app/lib/audit/checksum.rb

```diff
@@ -5,37 +5,27 @@ class << self
       delegate :logger, to: ::PreservationCatalog::Application
     end
 
+    # Queues asynchronous CV
     def self.validate_disk(endpoint_name, limit=Settings.c2m_sql_limit)
       logger.info "#{Time.now.utc.iso8601} CV validate_disk starting for #{endpoint_name}"
       # pres_copies is an AR Relation; it could return a lot of results, so we want to process it in
       # batches.  we can't use ActiveRecord's .find_each, because that'll disregard the order .fixity_check_expired
       # specified.  so we use our own batch processing method, which does respect Relation order.
       pres_copies = PreservedCopy.by_endpoint_name(endpoint_name).for_online_endpoints.fixity_check_expired
-      logger.info "Number of Preserved Copies to be checksum validated: #{pres_copies.count}"
-      ActiveRecordUtils.process_in_batches(pres_copies, limit) do |pc|
-        ChecksumValidator.new(pc).validate_checksums
-      end
+      logger.info "Number of Preserved Copies to be enqueued for CV: #{pres_copies.count}"
+      ActiveRecordUtils.process_in_batches(pres_copies, limit, &:validate_checksums!)
```

Comment:
actually, i think you don't want to keep using this method, because it expects synchronous processing of the records being batched:  https://github.com/sul-dlss/preservation_catalog/blob/92e1a0107aa7fc4a704690c013cc789812db2892/app/lib/active_record_utils.rb#L20-L40

if the checksum validation time isn't updated in the loop, it'll just process the first batch repeatedly.  i would just use `find_each`... i can't think of a great way to get batching that respects our `order by` without coming up with a more complex windowing scheme for the query than what we currently have implemented in that helper.

if we switch to `find_each` here, we should of course also ditch the no-longer-valid comment about using our own batching method.

Reply:
Ah, interesting point.  Happy to do that!